functorch/csrc/BatchRulesRandomness.cpp

```diff
@@ -149,6 +149,33 @@ Tensor unary_pointwise_random_batch_rule(const Tensor& tensor, ExtraArgs... extr
   return makeBatched(out, 0, cur_level);
 }
 
+template<typename... ExtraArgs>
+Tensor randint_like_random_batch_rule(const Tensor& self, ExtraArgs... extra_args) {
+  c10::impl::ExcludeDispatchKeyGuard guard(kVmapModeKey);
+  auto maybe_layer = maybeCurrentDynamicLayer();
+  const auto cur_level = maybe_layer->layerId();
+  const auto batch_size = maybe_layer->batchSize();
+  RandomnessType randomness = maybe_layer->randomness();
+  check_randomness(randomness);
+
+  Tensor tensor_value;
+  optional<int64_t> tensor_bdim;
+  std::tie(tensor_value, tensor_bdim) = unwrapTensorAtLevel(self, cur_level);
+  tensor_value = moveBatchDimToFront(tensor_value, tensor_bdim);
+
+  if (randomness == RandomnessType::Same && tensor_bdim) {
+    tensor_value = tensor_value[0];
+  } else if (randomness == RandomnessType::Different && !tensor_bdim) {
+    auto shape = tensor_value.sizes();
+    VmapDimVector shapeVec(shape.begin(), shape.end());
+    shapeVec.insert(shapeVec.begin(), maybe_layer->batchSize());
```

Comment:
Nit: this insertion is O(n), we should insert first, then copy over the rest of the list

Reply:
Also, shapeVec isn't being used?